### ISSUE TITLE
Hunter pets don't allow us to leave combat when an AOE ability hurts them + another target?

### ISSUE THREAD

**Issue by Mikik3jr** [Atlantiss/NetherwingBugtracker#2921]
[//]: # (Enclose links to things related to the bug using http://wowhead.com or any other TBC database.)
[//]: # (You can use screenshot ingame to visual the issue.)
[//]: # (Write your tickets according to the format:)
[//]: # ([Quest][Azuremyst Isle] Red Snapper - Very Tasty!)
[//]: # ([NPC] Magistrix Erona)
[//]: # ([Spell][Mage] Fireball)
[//]: # ([Npc][Drop] Ghostclaw Lynx)
[//]: # ([Web] Armory doesnt work)
[//]: # (To find server revision type `.server info` command in the game chat.)

**Description**: 
Several abilities mostly AOE abilities that hit a pet and another target can make a combat bug.

**Current behaviour**:
It got my attention that for example when a [Tempest-Forge Patroller](https://www.wowhead.com/npc=19166/tempest-forge-patroller#abilities) casts an AOE ability I am usually in combat bug. Only me (healer) and a warrior were bugged (shaman, mage, hunter wasn't). **The combat persisted until we made our hunter dissmiss their pet.** Maybe a warrior shout buffed the pet and I healed it and thats why only us were on combat.

I've just finished Ramparts HC, and the last boss were putting fire on the ground. I healed the hunter pet, and after the fight I was in combat, and ofc it was over after the pet disappeared. 

TL;DR: I saw that most of the time when an AOE ablity hits a hunter/warlock pet and another target, then the pet stays in combat + those whom had any interraction (heal, buff etc) with that pet.

**Expected behaviour**: 
Pets should leave combat after the fight ends, so they won't force us to stay in combat with them.

**Server Revision**: 
2345

**Comment by Wolffenstein** [comment]
Fixed in rev 2348.
